pkg/csidriver/driver.go

```diff
@@ -589,21 +601,33 @@ func mapBackingResourceToPod(dv *driverVolume) error {
 		// we can return the error back to volume provisioning, where the kubelet will retry at
 		// a controlled frequency
 		sharedSecret := client.GetSharedSecret(dv.GetSharedDataId())
+		if sharedSecret == nil {
+			klog.V(4).Infof("mapBackingResourceToPod for pod volume %s:%s:%s share %s no longer exists", dv.GetPodNamespace(), dv.GetPodName(), dv.GetVolID(), dv.GetSharedDataId())
+			return nil
+		}
 		sNamespace := sharedSecret.Spec.SecretRef.Namespace
 		sName := sharedSecret.Spec.SecretRef.Name
 		comboKey := objcache.BuildKey(sNamespace, sName)
-		s := client.GetSecret(sNamespace, sName)
-		if s != nil {
-			payload := Payload{
-				ByteData: s.Data,
-			}
 
-			upsertError := commonUpsertRanger(dv, comboKey, payload)
-			if upsertError != nil {
-				ProcessFileSystemError(s, upsertError)
-				return upsertError
+		s, err := client.GetSecret(sNamespace, sName)
+		if err != nil { // If the error is `Permission Denied, it returns a `PermissionDenied` error, which correctly fails the initial volume mount
+			if kerrors.IsForbidden(err) {
+				return status.Errorf(codes.PermissionDenied, "CSI driver does not have permission to get secret %s/%s", sNamespace, sName)
 			}
+			klog.Warningf("could not retrieve secret %s/%s: %v", sNamespace, sName, err)
+			return nil
```

Comment:
again don't eat this error and return nil ... return the error, or do another status.Errorf(codes.... with a generic error code